frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs

```diff
@@ -223,11 +273,70 @@ createAccountNotGasPayer ideL name chain mPublicKey selectedKeyset keyset = Work
         , createAccountSplash ideL name chain mPublicKey selectedKeyset <$ back
         )
 
+createAccountFromExternalAccount
+  :: forall t m mConf model key
+     . ( MonadWidget t m
+       , Monoid mConf
+       , HasNetwork model t
+       , HasNetworkCfg mConf t
+       , HasCrypto key (Performable m)
+       , HasCrypto key m
+       , HasLogger model t
+       , HasJsonData model t
+       , HasJsonDataCfg mConf t
+       , Flattenable mConf t
+       , HasWallet model key t
+       , HasWalletCfg mConf key t
+       , HasTransactionLogger m
+       )
+  => model
+  -> AccountName
+  -> ChainId
+  -> Maybe PublicKey
+  -> DefinedKeyset t
+  -> AccountGuard
+  -> Workflow t m (Text, (mConf, Event t ()))
+createAccountFromExternalAccount model name chain mPublicKey selectedKeyset keyset = Workflow $ do
+  receiveConf <- modalMain $
+    uiReceiveFromLegacy model
+
+  let
+    netInfo =
+      getNetworkInfoTriple $ model ^. network
+
+    isSubmitDisabled =
+      isNothing <$> _receiveFromLegacy_transferInfo receiveConf
+
+  modalFooter $ do
+    back <- cancelButton def "Back"
+    submit <- confirmButton (def & uiButtonCfg_disabled .~ isSubmitDisabled) "Submit"
+
+    let
+      nextWF = receiveFromLegacySubmitTransferCreate model never name chain
+        <$> _receiveFromLegacy_ttl receiveConf
+        <*> _receiveFromLegacy_gasLimit receiveConf
+
+      requiredInfo = (liftA2 .liftA2) (,) netInfo $
+        _receiveFromLegacy_transferInfo receiveConf
+
+      onSubmit = tagMaybe (current requiredInfo) submit
+
+      rewrapWF wfFn (netI, tfrI) = mapWorkflow ("Creating account",) $
+        wfFn netI tfrI keyset
+
+    pure ( ("Sender Details" , (_receiveFromLegacy_conf receiveConf, never) )
+         , leftmost
+           [ createAccountSplash model name chain mPublicKey selectedKeyset <$ back
```

Comment:
`mPublicKey selectedKeyset` are only used for rebuilding the widget we came from, better pass it in with the usual `self` binding trick.

Reply:
This causes the form to be cleared when you click 'back'